bigc/exceptions.py

```diff
@@ -1,6 +1,36 @@
+from __future__ import annotations
+
+from typing import Any, Optional
+
+from requests import Response
+import http
+
+
 class BigCommerceAPIException(Exception):
     """Base exception class for BigCommerce API errors"""
```

Comment:
```suggestion
class BigCommerceAPIException(Exception):
    """Base exception class for BigCommerce API errors"""

    response: Optional[Response]
```

Reply:
Does it? IMO type inference on class attributes is kind of iffy.

Reply:
<img width="916" alt="image" src="https://github.com/MedShift/bigc/assets/38477514/bbf166dd-66b9-4859-8aab-935dc83c360b">
VSC as well.

Reply:
What about `x.response = 'foo'`? Assignments are what I've had trouble with in the past.

Reply:
<img width="488" alt="image" src="https://github.com/MedShift/bigc/assets/38477514/25be9b4a-be4f-4a6d-b6d3-9d5b3fa60c28">
No error, but the typing is not respected by PyCharm even when adding the annotation
<img width="574" alt="image" src="https://github.com/MedShift/bigc/assets/38477514/ceedf547-7506-4ab8-84c0-0a4ef210dea1">